test/test.sh

```
@@ -13,25 +30,60 @@ int main(int argc, char **argv) {
-	rm -f cmp out a.out print_args
+	rm -f cmp out a.out print_args exit42 infinite_loop
 }
 
 assert() {
-	printf '%-50s:' "["$1]"
+	COMMAND="$1"
+	shift
+	printf '%-50s:' "[$COMMAND]"
 	# exit status
-	echo -n -e "$1" | bash >cmp 2>&-
+	echo -n -e "$COMMAND" | bash >cmp 2>&-
```

Shellcheck (0.11.0)

[warning] Redirecting to/from command name instead of file. Did you want pipes/xargs (or quote to ignore)?

(SC2238)